vapi/library/finder/finder.go

```diff
@@ -0,0 +1,309 @@
+/*
+Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package finder
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/vmware/govmomi/vapi/library"
+)
+
+// Finder is a helper object for finding content library objects by their
+// inventory paths: /LIBRARY/ITEM/FILE.
+//
+// Wildcard characters `*` and `?` are both supported. However, the use
+// of a wildcard character in the search string results in a full listing of
+// that part of the path's server-side items.
+//
+// Path parts that do not use wildcard characters rely on server-side Find
+// functions to find the path token by its name. Ironically finding one
+// item with a direct path takes longer than if a wildcard is used because
+// of the multiple round-trips. Direct paths will be more performant on
+// systems that have numerous items.
+type Finder struct {
+	M *library.Manager
```

Comment:
Hi @dougm,

I exported it in case someone wanted to create a `Finder` without the `NewFinder` function. The only reason it's named `M` is because I loathe long lines of code, but sure, it can be renamed.